torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -269,10 +248,11 @@ def _share_state_and_init_handle_attrs(
         fsdp_state._default_stream = root_state._default_stream
         fsdp_state._exec_order_data = root_state._exec_order_data
         fsdp_state._free_event_queue = root_state._free_event_queue
-        fsdp_state._device_mesh = root_state._device_mesh
         handle = fsdp_state._handle
         if handle:
             handle.init_flat_param_attributes()
+        if hasattr(root_state, "_device_mesh"):
```

Comment:
Why do we need `hasattr`? Shouldn't `_device_mesh` always exist but can be `None`?

Reply:
> Why do we need `hasattr`? Shouldn't `_device_mesh` always exist but can be `None`?

If I remember correctly, this is because it would break a composable API unit test. 
I can add a TODO here to update in later PR when we add device_mesh to composable APIs.